sdgym/datasets.py

```diff
@@ -45,9 +47,24 @@ def _get_dataset_path(dataset, datasets_path, bucket=None, aws_key=None, aws_sec
     return dataset_path
 
 
-def load_dataset(dataset, datasets_path=None, bucket=None, aws_key=None, aws_secret=None):
+def load_dataset(dataset, datasets_path=None, bucket=None, aws_key=None, aws_secret=None,
+                 max_columns=None):
     dataset_path = _get_dataset_path(dataset, datasets_path, bucket, aws_key, aws_secret)
-    metadata = Metadata(str(dataset_path / 'metadata.json'))
+    metadata_file = open(str(dataset_path / 'metadata.json'))
```

Comment:
This should be done inside a `with` block instead, so the file is closed after we are done reading it.
Also, I think the `str` call is not needed here.